### PR TITLE
maven plugin IT optimization

### PR DESCRIPTION
**Pull request by LorenzoBettini**
This PR aims at optimizing the IT tests for our Maven plugins in two main respects:

- when building our IT projects, we also set the local Maven repository cache (by default, ~/.m2/repository, filled up to this point of the build) as a Maven repository to speed up dependency resolution when building the IT projects (Inspired by https://maven.apache.org/plugins/maven-invoker-plugin/examples/fast-use.html). Note that `~/.m2/repository` is used as if it was a "remote" repository in read-only mode when running IT projects: it won't be corrupted.
- instead of copying the IT projects to the OS temporary directory, we copy them into the `target` folder (subdirectory `local-test-dir`). This will make the inspection of logs much easier. Moreover, that directory is wiped before running the tests, but NOT after, so that their contents
can be inspected after the tests.

Concerning the second point, I had to refactor Xtend Maven plugin ITs a lot because especially compiler tests were continuously removing previously executed project folders, making it impossible to inspect the build logs. In that respect, to make the test for `symlinks` work in this new setup, I created another copy of `multisources`, `multisources-to-link`, because `multisources` was also used in another test and I didn't want that to be removed when running `symlinks` project.

### CODE REVIEW

**Review by cdietrich — APPROVED** [review]

**Comment by LorenzoBettini** [comment]
I'll wait for @szarnekow review

**Comment by LorenzoBettini** [comment]
@szarnekow can we merge this?

**Review by szarnekow — APPROVED** [review]
Yes, looks good.

File: org.eclipse.xtend.maven.plugin/Run ITs from Eclipse.launch [hunk 1]
```diff
@@ -12,11 +12,12 @@
...
+    <booleanAttribute key="org.eclipse.jdt.launching.ATTR_SHOW_CODEDETAILS_IN_EXCEPTION_MESSAGES" value="true"/>
     <booleanAttribute key="org.eclipse.jdt.launching.ATTR_USE_CLASSPATH_ONLY_JAR" value="false"/>
     <booleanAttribute key="org.eclipse.jdt.launching.ATTR_USE_START_ON_FIRST_THREAD" value="true"/>
     <stringAttribute key="org.eclipse.jdt.launching.CLASSPATH_PROVIDER" value="org.eclipse.m2e.launchconfig.classpathProvider"/>
     <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value=""/>
     <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="org.eclipse.xtend.maven.plugin"/>
     <stringAttribute key="org.eclipse.jdt.launching.SOURCE_PATH_PROVIDER" value="org.eclipse.m2e.launchconfig.sourcepathProvider"/>
-    <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-ea -DtestMavenRepo=${project_loc:org.eclipse.xtend.maven.plugin}/target/.m2test"/>
+    <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-ea -DtestMavenRepo=${project_loc:org.eclipse.xtend.maven.plugin}/target/.m2test -Dmaven.test.tmpdir=${project_loc:org.eclipse.xtend.maven.plugin}/target/local-test-dir"/>
```
[the remark this answers is not in the document]
Comment (LorenzoBettini) [reply org.eclipse.xtend.maven.plugin/Run ITs from Eclipse.launch:22]: This corresponds to the same property in the POM. When we run it from Eclipse we cannot benefit from the property in the POM, and we must pass it in the launch configuration.

**Comment by LorenzoBettini** [comment]
@szarnekow thanks! I rebased on top of main and if... ehm... when the build is green I'll merge ;)

**Pull request merged**